builtInFunctions/esdtMetaDataRecreate.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+	"math/big"
+)
+
+const newAttributesIndex = 2
+
+type esdtMetaDataRecreate struct {
+	baseActiveHandler
+	globalSettingsHandler vmcommon.GlobalMetadataHandler
+	storageHandler        vmcommon.ESDTNFTStorageHandler
+	rolesHandler          vmcommon.ESDTRoleHandler
+	accounts              vmcommon.AccountsAdapter
+}
+
+// NewESDTMetaDataRecreateFunc returns the esdt meta data recreate built-in function component
+func NewESDTMetaDataRecreateFunc(
+	accounts vmcommon.AccountsAdapter,
+	globalSettingsHandler vmcommon.GlobalMetadataHandler,
+	storageHandler vmcommon.ESDTNFTStorageHandler,
+	rolesHandler vmcommon.ESDTRoleHandler,
+	enableEpochsHandler vmcommon.EnableEpochsHandler,
+) (*esdtMetaDataRecreate, error) {
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+	if check.IfNil(globalSettingsHandler) {
+		return nil, ErrNilGlobalSettingsHandler
+	}
+	if check.IfNil(enableEpochsHandler) {
+		return nil, ErrNilEnableEpochsHandler
+	}
+	if check.IfNil(storageHandler) {
+		return nil, ErrNilESDTNFTStorageHandler
+	}
+	if check.IfNil(rolesHandler) {
+		return nil, ErrNilRolesHandler
+	}
+
+	e := &esdtMetaDataRecreate{
+		accounts:              accounts,
+		globalSettingsHandler: globalSettingsHandler,
+		storageHandler:        storageHandler,
+		rolesHandler:          rolesHandler,
+	}
+
+	e.baseActiveHandler.activeHandler = enableEpochsHandler.IsDynamicESDTEnabled
+
+	return e, nil
+}
+
+// ProcessBuiltinFunction saves the token type in the system account
+func (e *esdtMetaDataRecreate) ProcessBuiltinFunction(acntSnd, _ vmcommon.UserAccountHandler, vmInput *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	if vmInput == nil {
+		return nil, ErrNilVmInput
+	}
+	if vmInput.CallValue == nil {
+		return nil, ErrNilValue
+	}
+	if vmInput.CallValue.Cmp(zero) != 0 {
+		return nil, ErrBuiltInFunctionCalledWithValue
+	}
+	if len(vmInput.Arguments) != 3 {
+		return nil, ErrInvalidNumberOfArguments
+	}
+	if !bytes.Equal(vmInput.CallerAddr, vmInput.RecipientAddr) {
+		return nil, ErrInvalidRcvAddr
+	}
+	if check.IfNil(acntSnd) {
+		return nil, ErrNilUserAccount
+	}
+	if !e.baseActiveHandler.IsActive() {
+		return nil, ErrBuiltInFunctionIsNotActive
+	}
+	// TODO check and consume gas
+
+	err := e.rolesHandler.CheckAllowedToExecute(acntSnd, vmInput.Arguments[tokenIDIndex], []byte(core.ESDTRoleModifyCreator))
+	if err != nil {
+		return nil, err
+	}
+
+	esdtTokenKey := append([]byte(baseESDTKeyPrefix), vmInput.Arguments[tokenIDIndex]...)
+	nonce := big.NewInt(0).SetBytes(vmInput.Arguments[nonceIndex]).Uint64()
+	esdtData, err := e.storageHandler.GetESDTNFTTokenOnSender(acntSnd, esdtTokenKey, nonce)
+	if err != nil {
+		return nil, err
+	}
+	if esdtData.Type != uint32(core.DynamicNFT) {
+		return nil, ErrOperationNotPermitted
+	}
+
+	esdtData.TokenMetaData.Attributes = vmInput.Arguments[newAttributesIndex]
```

Comment:
this is not only about attributes. Check NFT Create function where you receive a set of arguments for URI, Name, Attributes, Royalties and so forth.

This is exactly like NFT create. but it does not create a new nonce.

Reply:
Done.